vars/stageDeployToOpenshift.groovy

```diff
@@ -12,6 +12,28 @@ def call(def context) {
       destTag: "latest",
       namespace: context.targetProject
     )
+
+    def ocpLatestImage = sh(returnStdout: true, script:"oc get istag ${context.componentId}:latest -n ${context.targetProject} -o jsonpath='{.image.dockerImageReference}'", label : "find last image").trim()
+      
+    def ocpDeployment
+    timeout(context.openshiftBuildTimeout) {
+      while(true) {
+        ocpDeployment = sh(returnStdout: true, script:"sleep 10 && oc describe dc ${context.componentId} -n ${context.targetProject} | grep -e ${ocpLatestImage} -e Status -e 'Latest Version'", label : "find new deployment").trim().split(/\s+/)
+        
+        echo ("Found last deployment id: ${ocpDeployment[2]} - status ${ocpDeployment[6]}")
+        if (ocpDeployment[6] != "Pending" && ocpDeployment[6] != "Running") {
+            echo "OCP Deployment done - reporting status"
+            break
+        }
+      }
```

Comment:
I will try to come up with a different solution for this which does not rely on `sleep 10`. But we can merge this like it is here and then improve (if that is possible at all).